libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt

```diff
@@ -0,0 +1,87 @@
+package com.mapbox.navigation.base.internal.route
+
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import com.mapbox.api.directions.v5.models.MaxSpeed
+import kotlin.math.min
+
+internal object AnnotationsRefresher {
```

Comment:
just curious, why do you need an object if you can just define a function without it? 🤔

Reply:
Mostly I introduced an object to simplify testing. Actually, this object has several methods. Moreover, it does one particular not the simplest thing: merges the annotations from old leg to a new one. Why shouldn't it be in a separate class?